src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/EvidenceTargetLinkClusterer.java

```diff
@@ -0,0 +1,122 @@
+                            nextEvidence.getDistalTargets(readMetadata).get(0),
+                            nextEvidence.getDistalTargetStrands(readMetadata).get(0),
+                            nextEvidence instanceof BreakpointEvidence.DiscordantReadPairEvidence
+                                    ? 0 : 1,
+                            nextEvidence instanceof BreakpointEvidence.DiscordantReadPairEvidence
+                                    ? 1 : 0);
+                }
+                currentIntervalsWithTargets.put(updatedLink.source, updatedLink);
+            }
+        }
+
+        for (Iterator<SVIntervalTree.Entry<EvidenceTargetLink>> it = currentIntervalsWithTargets.iterator(); it.hasNext(); ) {
+            links.add(it.next().getValue());
+        }
+
+        return links.iterator();
+    }
+
+    private static boolean strandsMatch(final Boolean forwardStrand1, final Boolean forwardStrand2) {
+        return forwardStrand1 != null && forwardStrand2 != null && forwardStrand1.equals(forwardStrand2);
+    }
+
+    /**
+     * Combines links that agree on interval-pair and orientation but have source and target switched. For
+     * example, if link1 has intervals s1 and t1, and link2 has s2 and t2, and s1 overlaps t1 and s2 overlaps t2,
+     * the links will be combined as long as strands agree. Returned links have the intersection of two paired intervals as
+     * source and target, with the lower-coordinate interval appearing as source.
+     */
+    public static List<EvidenceTargetLink> deduplicateTargetLinks(final List<EvidenceTargetLink> evidenceTargetLinks) {
+
+        PairedStrandedIntervalTree<EvidenceTargetLink> pairedStrandedIntervalTree = new PairedStrandedIntervalTree<>();
+
+        evidenceTargetLinks.stream().filter(link -> link.source.compareTo(link.target) < 0).forEach(link -> pairedStrandedIntervalTree.put(link.getPairedStrandedIntervals(), link));
+
+
+        evidenceTargetLinks.stream().filter(link -> link.source.compareTo(link.target) >= 0).forEach(link -> {
+
+            PairedStrandedIntervals reversedPair =
+                    new PairedStrandedIntervals(link.target, link.targetForwardStrand, link.source, link.sourceForwardStrand);
+
+            Iterator<Tuple2<PairedStrandedIntervals, EvidenceTargetLink>> psiOverlappers =
+                    pairedStrandedIntervalTree.overlappers(reversedPair);
+
+            EvidenceTargetLink newLink = null;
+            while (psiOverlappers.hasNext()) {
+                EvidenceTargetLink existingLink = psiOverlappers.next()._2();
+                newLink = new EvidenceTargetLink(link.target.intersect(existingLink.source), link.targetForwardStrand,
+                        link.source.intersect(existingLink.target), link.sourceForwardStrand,
+                        link.splitReads + existingLink.splitReads, Math.max(link.readPairs, existingLink.readPairs));
+                psiOverlappers.remove();
+                break;
```

Comment:
Is this right?  You intended to break here?
If so, the while on line 102 should just be an if, but I'm guessing you intended to process all the overlappers rather than just the first.

Reply:
I realized that way I'm setting up the data that I'm calling this on, there can never be more than one overlapper. I'll either fix this to make it more clear or re-do it if I decide to take your suggestion to canonicalize the links.